packages/react-core/src/components/SearchInput/SearchInput.tsx

```diff
@@ -21,6 +21,15 @@ export interface SearchAttribute {
   display: React.ReactNode;
 }
 
+export interface expandableProps {
+  /** Flag to indicate if the search input is expanded */
+  isExpanded: boolean;
+  /** Callback function to toggle the expandable search input */
+  onToggleExpand: (isExpanded: boolean, event: React.SyntheticEvent<HTMLButtonElement>) => void;
+  /** An accessible label for the expandable search input toggle */
+  toggleAriaLabel?: string;
```

Comment:
Because an a11y error will get flagged for the icon button not having an accessible name, we should make this a required prop as well.